### PR TITLE
Add name-generation test

### PR DESCRIPTION
**Pull request by cevich**
Podman's container-name generation depends on the global RNG state being properly initialized (seeded).  Should this not happen for some reason (or it's seeded with a static value), podman will generate the exact same repeating sequence of container names (assuming no clashes with existing containers).  Add a test to confirm this is always the case.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by mtrmac — COMMENTED** [review]
Thanks for following up.

Code LGTM (noting that I am not intimately familiar with the `podmanTest` infrastructure and I didn’t study that carefully.)

---

Highlighting for the record and possible discussion: This might be a flaky test.

The name generator seems capable of creating something on  the order of 25 740 names (+- a few), so that's 1/25740 chance of the test flaking. You probably know how many times in a day/month/year that is.

**Review by edsantiago — APPROVED** [review]
I'm the kind of person who really loves comments, such as a top-level comment "this test confirms that RNG seeding is working; if that breaks, we get the same seed and the same name on each run; see #ISSUE", but I won't block over that. I just like being considerate to future maintainers who will see this and wonder wtf.

And I agree with @mtrmac, it seems Good Enough to just test podman; the chances of this breaking on remote-only seem slim enough to me.

And, note for future, under ginkgov2 this can have a Flake decorator which will retry, and of course at that time one can add a comment explaining the namespace concern.

**Comment by mtrmac** [comment]
> such as a top-level comment

Oh, great point. Yes please.

**Comment by cevich** [comment]
> Oh, great point. Yes please.

Isn't this the purpose of the `It("blah")` line?  Looking through the e2e tests, basically none of them have a top-level comment (or many comments at all for that matter) :disappointed: So...errr...at least the new test is consistent :rofl: 

I did put a good description in the commit, which git-blame will point people directly at.  That's got to be worth something :grin:

**Comment by mtrmac** [comment]
My fault for not reading that, that’s indeed a good place.

**Comment by edsantiago** [comment]
Thing is, most tests make sense. This one is baffling: as a human reader, it seems completely pointless. When something seems pointless, sometimes it _is_ (leftovers from prior era), sometimes it _isn't_ (very good reason, not obvious at first sight). Nobody wants to di `git blame`. As a maintainer, if someone forces me to do that, I'm going to resent them. We have a pretty neat technology, in which a developer can write explanations right next to the actual code, without incurring a compile-time cost, nor run-time, nor code bloat. It's something that takes a few seconds to add, and can save future maintainers many minutes of headscratching.

But, shrug, I can add the comments in the future when we do ginkgov2 and the Flake thing. (I'll resent it, though).

File: test/e2e/run_test.go
Comment (Luap99) [reply test/e2e/run_test.go:52]: As someone who spend most of their time in the last weeks in the e2e test I find it always very confusing to add skips for no good reason? There is no need for the test to use `--latest`.

I see @mtrmac points that is not necessary but I personally prefer to have as little skips as possible.

File: test/e2e/run_test.go
Comment (Luap99) [inline test/e2e/run_test.go:55]: maybe a bit better golang style would be to use
```suggestion
		for i := range names {
```

File: test/e2e/run_test.go
Comment (Luap99) [inline test/e2e/run_test.go:70]:
```suggestion
		Expect(names[0]).ToNot(Equal(names[1]), "names are equal, is the RNG seeded correctly?")
```

A description to the actual error will go a long way if this ever is failing.

Also isn't this whole test flaky by default? I mean technically speaking there is nothing wrong with returning the same name twice. Sure it is unlikely, I looked at the code and there are `25487` possible names, so the chance of this picking the same name is 1/25487.

Comment (cevich) [reply test/e2e/run_test.go:70]: There's two names being generated, so I believe the odds are more like 1/25487^2 but my statistics expertise is at least 30 years old :grin:

File: test/e2e/run_test.go
Comment (cevich) [reply test/e2e/run_test.go:52]: You're right, these skips tend to hand around for years and years.  I'll remove it.

**Comment by cevich** [comment]
Force-push: Addressed all feedback.

**Comment by cevich** [comment]
Force-push: Minor comment tweak.

**Review by Luap99 — APPROVED** [review]
/lgtm
/hold

**Comment by cevich** [comment]
force-push: Typo fix.

**Comment by cevich** [comment]
Force-push: Attempt to fix empty `cid` value from [previous test failure](https://api.cirrus-ci.com/v1/artifact/task/6354557333143552/html/int-podman-fedora-38-root-host-sqlite.log.html#t--podman-generates-different-names-for-successive-containers--1).

File: test/e2e/run_test.go
Comment (cevich) [inline test/e2e/run_test.go:62]: @Luap99 any idea why `--cidfile=/dev/stdout` [isn't working](https://api.cirrus-ci.com/v1/artifact/task/6596184182095872/html/int-podman-debian-12-root-host-boltdb.log.html#t--podman-generates-different-names-for-successive-containers--3) (log)?  It seems to work fine for me locally (F37).

Comment (cevich) [reply test/e2e/run_test.go:62]: Maybe I should use `/proc/self/fd/1` instead?

Comment (Luap99) [reply test/e2e/run_test.go:62]: just use `podman create`, faster and will cause less flakes. I don't know why this is not working, the output capturing in ginkgo is a bit special. Interestingly the log shows it is working in the first iteration but then the same command fails in the second iteration.

Comment (cevich) [reply test/e2e/run_test.go:62]: Good idea, thanks Paul.

**Comment by cevich** [comment]
Force-push: s/podman run/podman create/

**Comment by edsantiago** [comment]
/lgtm

If we start seeing flakes, one possible fix is to run ~ten creates, store results in a map, and `Expect(numberofkeys).To(<3)`

**Comment by cevich** [comment]
> If we start seeing flakes

@mtrmac or @Luap99 follow-up to the flake concern:  Is there anything else besides names that uses the global RNG?  Like container or image IDs or IP addresses, something with a larger pool of choices?

**Comment by cevich** [comment]
> That doesn’t seem worthwhile.

I agree, but thanks for taking a look and suggesting it.  It's probably a good thing only the name-generator depends on the global state anyway.  Thanks @mtrmac

**Pull request merged**
